### DISCUSSION TITLE
gas_assembly's coolant channel positions

### DISCUSSION THREAD

**Discussion by Kevrog1006** [neams-th-coe/cardinal#1003] [Q&A] [answered]
Hello

I want to do some work related to this example.
But I didn 't find the relevant instructions about the file named coolant_channel_positions.txt. 
What should I do if I want to get the parameter data from my solidmesh file or assembly.py about this file? 

thanks a lot!

**Comment by aprilnovak** [comment, marked as answer]
Hi @Kevrog1006, the `coolant_channel_positions.txt` file was actually generated by myself with a python script that computes the centers of those pins. At the time the tutorial was created, that was the only option to obtain those coordinates.

However, there is a new system in MOOSE called the `Positions` system that allows you to extract this information programmatically. I have not had the chance to use it before, but there is some documentation here: https://mooseframework.inl.gov/syntax/Positions/. I believe @GiudGiud has designed a lot of that system, so perhaps he can help with any specifics.

**Reply by GiudGiud** [reply]
if you already have the file you can use a FilePositions to load it, then retrieve a pointer to the positions object in the object you are developing and that lets you call things like "which point is nearest" from the positions object

if you dont, as long as the points are distributed like something in that list it's already implemented. if not, happy to help
https://mooseframework.inl.gov/syntax/Positions/index.html
